src/settings/global_user.rs

```diff
@@ -33,80 +129,122 @@ impl From<GlobalUser> for Credentials {
+pub fn default_config_file() -> Result<PathBuf, failure::Error> {
     let home_dir = if let Ok(value) = env::var("WRANGLER_HOME") {
-        info!("Using WRANGLER_HOME: {}", value);
+        log::info!("Using WRANGLER_HOME: {}", value);
         Path::new(&value).to_path_buf()
     } else {
-        info!("No WRANGLER_HOME detected");
+        log::info!("No WRANGLER_HOME detected");
         dirs::home_dir()
             .expect("oops no home dir")
             .join(".wrangler")
     };
-    let global_config_dir = home_dir.join("config");
-    info!("Using global config dir: {:?}", global_config_dir);
-    Ok(global_config_dir)
+    let global_config_file = home_dir.join("config").join(DEFAULT_CONFIG_FILE_NAME);
+    log::info!("Using global config file: {}", global_config_file.display());
+    Ok(global_config_file)
 }
 
 #[cfg(test)]
 mod tests {
     use super::*;
-    use std::env;
+    use std::fs::File;
+    use tempfile::tempdir;
+
+    use crate::settings::environment::MockEnvironment;
 
     #[test]
     fn it_can_prioritize_token_input() {
         // Set all CF_API_TOKEN, CF_EMAIL, and CF_API_KEY.
         // This test evaluates whether the GlobalUser returned is
         // a GlobalUser::TokenAuth (expected behavior; token
         // should be prioritized over email + global API key pair.)
-        env::set_var("CF_API_TOKEN", "foo");
-        env::set_var("CF_EMAIL", "test@cloudflare.com");
-        env::set_var("CF_API_KEY", "bar");
+        let mut mock_env = MockEnvironment::default();
+        mock_env.set(CF_API_TOKEN, "foo");
+        mock_env.set(CF_EMAIL, "test@cloudflare.com");
+        mock_env.set(CF_API_KEY, "bar");
 
-        let user = get_global_config().unwrap();
+        let tmp_dir = tempdir().unwrap();
+        let config_dir = test_config_dir(&tmp_dir, None).unwrap();
+
+        let user = GlobalUser::build(mock_env, config_dir).unwrap();
         assert_eq!(
             user,
             GlobalUser::TokenAuth {
                 api_token: "foo".to_string()
             }
         );
     }
+
+    #[test]
+    fn it_can_prioritize_env_vars() {
+        let api_token = "thisisanapitoken";
+        let api_key = "reallylongglobalapikey";
+        let email = "user@example.com";
+
+        let file_user = GlobalUser::TokenAuth {
+            api_token: api_token.to_string(),
+        };
+        let env_user = GlobalUser::GlobalKeyAuth {
+            api_key: api_key.to_string(),
+            email: email.to_string(),
+        };
+
+        let mut mock_env = MockEnvironment::default();
+        mock_env.set(CF_EMAIL, email);
+        mock_env.set(CF_API_KEY, api_key);
+
+        let tmp_dir = tempdir().unwrap();
+        let tmp_config_path = test_config_dir(&tmp_dir, Some(file_user)).unwrap();
+
+        let new_user = GlobalUser::build(mock_env, tmp_config_path).unwrap();
+
+        assert_eq!(new_user, env_user);
+    }
+
+    #[test]
+    fn it_falls_through_to_config_with_no_env_vars() {
```

Comment:
Let's add a test to verify that environment variables work when no config directory is set up (the CI use case)